parser.cpp

```diff
@@ -1073,8 +1073,15 @@ namespace Sass {
       (*map) << make_pair(key, value);
     }
 
-    if (map->has_duplicate_key())
-    { error("Duplicate key \"" + map->get_duplicate_key()->perform(&to_string) + "\" in map " + map->perform(&to_string) + ".", pstate); }
+    // Check was moved to eval step
+    // if (map->has_duplicate_key()) {
+    //   To_String to_string(&ctx);
+    //   error("Duplicate key \"" + map->get_duplicate_key()->perform(&to_string) + "\" in map " + map->perform(&to_string) + ".", pstate);
+    // }
+
+    ParserState ps = map->pstate();
+    ps.offset = pstate - ps + pstate.offset;
+    map->pstate(ps);
```

Comment:
The check for duplicates keys was moved to the eval stage. The reason for this is, ie. when we have variables as keys, we would check like `$a` == `$b` for duplicate keys. But we actually need to check it only after the keys have been evaluated ... I was not sure if we really can get rid of the first check, but it seems so ...